components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/storage/model/BundleEntity.java

```diff
@@ -0,0 +1,52 @@
+package com.sap.cloud.cmp.ord.service.storage.model;
+
+import org.eclipse.persistence.annotations.Convert;
+import org.eclipse.persistence.annotations.TypeConverter;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+@Entity(name = "bundle")
+@Table(name = "bundles")
+public class BundleEntity {
+    @javax.persistence.Id
+    @Column(name = "id")
+    @Convert("uuidConverter")
+    @TypeConverter(name = "uuidConverter", dataType = Object.class, objectType = UUID.class)
+    private UUID Id;
+
+    @Column(name = "ord_id", length = 256)
+    @NotNull
+    private String ordId;
+
+    @Column(name = "name", length = 256)
+    @NotNull
+    private String title;
+
+    @Column(name = "short_description", length = 255)
+    private String shortDescription;
+
+    @Column(name = "description", length = Integer.MAX_VALUE)
+    private String description;
+
+    @ElementCollection
+    @CollectionTable(name = "links", joinColumns = @JoinColumn(name = "bundle_id"))
+    private List<Link> links;
+
+    @ElementCollection
+    @CollectionTable(name = "ord_labels", joinColumns = @JoinColumn(name = "bundle_id"))
+    private List<Label> labels;
+
+    @ElementCollection
+    @CollectionTable(name = "credential_request_strategies", joinColumns = @JoinColumn(name = "bundle_id"))
+    private List<CredentialsRequestStrategy> credentialsRequestStrategies;
+
+    @OneToMany(mappedBy = "bundleEntity", fetch = FetchType.LAZY)
+    private Set<APIEntity> apis;
+
+    @OneToMany(mappedBy = "bundleEntity", fetch = FetchType.LAZY)
```

Comment:
These relationships are currently specified as bidirectional, what is the reason for that ? Would expanding from both sides work if they were unidirectional ?

Reply:
What do you mean by bidirectional? It is @OneToMany relation and this is how JPA expects this to be expressed declaratively, in the Bundle you say that you have a collection of APIs and what is the field in the API is responsible for this relation (this is needed because the foreign-key is in the API and you need to specify there which column is the foreign key and since it is not in the Bundle cannot be expressed in the Bundle Entity)

Reply:
bidirectional means that you have @OneToMany here and @ManyToOne on the other side

Reply:
I got it, the odata-jpa library seems to need it